packages/jest-config/src/normalize.ts

```diff
@@ -1057,6 +1061,13 @@ export default function normalize(
     newOptions.logHeapUsage = false;
   }
 
+  if (newOptions.collectCoverage && newOptions.v8Coverage) {
+    throw createConfigError(
+      `  Configuration options ${chalk.bold('collectCoverage')} and` +
+        ` ${chalk.bold('v8Coverage')} cannot be used together.`,
```

Comment:
yargs has a "conflicts" method, for describing exclusive arguments; I wonder if we could find a way to make it work for your use-case, e.g., accepting an additional parameter which could be an Error instance.